src/models/Keyframes.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+import StyleSheet from '../models/StyleSheet'
+
+export default class Keyframes {
+  id: string
+  name: string
+  rules: Array<string>
+
+  constructor(name: string, rules: Array<string>) {
+    this.name = name
+    this.rules = rules
+
+    this.id = `sc-keyframes-${name}`
+  }
+
+  inject = (styleSheet: StyleSheet) => {
+    if (!styleSheet.hasNameForId(this.id, this.name)) {
+      styleSheet.inject(this.id, this.rules, this.name)
+    }
+  }
+
+  getName() {
```

Comment:
Just an idea; if we implement “valueOf” and “toString” to inject the keyframes CSS, then the effects of this breaking change would be minimal :) cc @probablyup

Reply:
I wrote about it https://github.com/styled-components/styled-components/pull/1930#issuecomment-414242846

And in previous PR you said that explicit coercion is better than implicit ;)

Reply:
Honestly, I feel like it's fine for this to be a breaking change in v4—let's not do too much hacky stuff?

Reply:
Sorry, just thinking about breaking changes again, since we already have quite a lot 😅 In this case, just refer to my old comment lol. Sorry about that.

@mxstbr @probablyup If we can get this change in, we can probably deprecate a couple of old uses of the StyleSheet and get rid of cloning soon, since we won’t have non-component usages anymore